app/controllers/events_controller.rb

```diff
@@ -0,0 +1,55 @@
+require 'integrations'
+require 'payload_validator'
+
+class EventsController < ApplicationController
+  SIGNING_KEY = ENV.fetch('INTEGRATIONS_SIGNING_KEY').freeze
+  EVENTS = YAML.load File.read(Rails.root.join('data', 'events.yml')).freeze
+
+  before_action :verify_signature, only: [:create]
+
+  def index
+    render json: EVENTS
+  end
+
+  def create
+    begin
+      body = MultiJson.load(request.body.read, symbolize_keys: true)
+      unless %i(event_type integrations payload).all? { |key| body.key? key }
+        return invalid_request
+      end
+
+      Integrations.send_event(body)
+      render json: { status: 'ok' }, status: :created
+    rescue MultiJson::ParseError
+      invalid_request('unable to parse request', type: 'parse_error')
+    rescue Integrations::UnsupportedIntegrationError => e
```

Comment:
Can you just catch ``Integrations`` and have the exception class have the error message + type? See https://github.com/rainforestapp/Rainforest/blob/2a720dc1780e6c9a3ab2c43f81f5b192ac3c3c25/lib/work.rb#L432 for examples.